libs/libc/stream/lib_blkoutstream.c

```diff
@@ -239,18 +239,18 @@ void lib_blkoutstream_close(FAR struct lib_blkoutstream_s *stream)
 {
   if (stream != NULL)
     {
-      if (stream->inode != NULL)
-        {
-          close_blockdriver(stream->inode);
-          stream->inode = NULL;
-        }
-
       if (stream->cache != NULL)
         {
           blkoutstream_flush(&stream->common);
           lib_free(stream->cache);
           stream->cache = NULL;
         }
+
+      if (stream->inode != NULL)
+        {
+          close_blockdriver(stream->inode);
+          stream->inode = NULL;
+        }
```

Comment:
Please include comment message to this commit. What was the site effect before this patch? The last data in cache was lost?